src/styles/withStyles.js

```diff
@@ -1,303 +1,19 @@
-      }
-
-      sheetManagerTheme.refs += 1;
-    }
-
-    detach(theme) {
-      if (this.disableStylesGeneration) {
-        return;
-      }
-
-      const stylesCreatorSaved = this.stylesCreatorSaved;
-      const sheetManager = this.sheetsManager.get(stylesCreatorSaved);
-      const sheetManagerTheme = sheetManager.get(theme);
-
-      sheetManagerTheme.refs -= 1;
-
-      if (sheetManagerTheme.refs === 0) {
-        sheetManager.delete(theme);
-        this.jss.removeStyleSheet(sheetManagerTheme.sheet);
-        const sheetsRegistry = this.context[ns.sheetsRegistry];
-        if (sheetsRegistry) {
-          sheetsRegistry.remove(sheetManagerTheme.sheet);
-        }
-      }
-    }
-
-    unsubscribeId = null;
-    jss = null;
-    sheetsManager = sheetsManager;
-    disableStylesGeneration = false;
-    stylesCreatorSaved = null;
-    theme = null;
-    sheetOptions = null;
-    theme = null;
-
-    render() {
-      const { classes: classesProp, innerRef, ...other } = this.props;
-
-      let classes;
-      let renderedClasses = {};
-
-      if (!this.disableStylesGeneration) {
-        const sheetManager = this.sheetsManager.get(this.stylesCreatorSaved);
-        const sheetsManagerTheme = sheetManager.get(this.theme);
-        renderedClasses = sheetsManagerTheme.sheet.classes;
-      }
-
-      if (classesProp) {
-        classes = {
-          ...renderedClasses,
-          ...Object.keys(classesProp).reduce((accumulator, key) => {
-            warning(
-              renderedClasses[key] || this.disableStylesGeneration,
-              [
-                `Material-UI: the key \`${key}\` ` +
-                  `provided to the classes property is not implemented in ${getDisplayName(
-                    Component,
-                  )}.`,
-                `You can only override one of the following: ${Object.keys(renderedClasses).join(
-                  ',',
-                )}`,
-              ].join('\n'),
-            );
-
-            warning(
-              !classesProp[key] || typeof classesProp[key] === 'string',
-              [
-                `Material-UI: the key \`${key}\` ` +
-                  `provided to the classes property is not valid for ${getDisplayName(Component)}.`,
-                `You need to provide a non empty string instead of: ${classesProp[key]}.`,
-              ].join('\n'),
-            );
-
-            if (classesProp[key]) {
-              accumulator[key] = `${renderedClasses[key]} ${classesProp[key]}`;
-            }
-
-            return accumulator;
-          }, {}),
-        };
-      } else {
-        classes = renderedClasses;
-      }
-
-      const more = {};
-
-      // Provide the theme to the wrapped component.
-      // So we don't have to use the `withTheme()` Higher-order Component.
-      if (withTheme) {
-        more.theme = this.theme;
-      }
-
-      return <Component classes={classes} {...more} {...other} ref={innerRef} />;
-    }
-  }
-
   Style.propTypes = {
     /**
```

Comment:
Do we need this?

Reply:
I mean, the propTypes ?

Reply:
`innerRef` is exclusive to `withStyles`, classes though can be omitted.